source/h2_stream.c

```diff
@@ -261,22 +288,36 @@ static void s_stream_cross_thread_work_task(struct aws_channel_task *task, void
         /* window_update_size is ensured to be not greater than AWS_H2_WINDOW_UPDATE_MAX */
         window_update_size = stream->synced_data.window_update_size;
         stream->synced_data.window_update_size = 0;
+        requested_reset_error_code = stream->synced_data.requested_reset_error_code;
 
         s_unlock_synced_data(stream);
     } /* END CRITICAL SECTION */
 
     while (!aws_linked_list_empty(&pending_frames)) {
         struct aws_linked_list_node *node = aws_linked_list_pop_front(&pending_frames);
```

Comment:
all the frames here are window update for now. But, who knows what will we do in the future. (Probably let stream take care of all the frames about stream?